services/store.go

```diff
@@ -1050,6 +1030,89 @@ func decodeTransfer(height uint32, info *event.ExecuteNotify, evt *event.NotifyE
 	return xfer
 }
 
+func checkEvmEventLog(evt *event.NotifyEventInfo) (bool, *ctypes.StorageLog) {
+	ethLog, err := event.NotifyEventInfoToEvmLog(evt)
+	if err != nil {
+		return false, nil
+	}
+	return true, ethLog
+}
+func parseEvmOngTransferLog(ethLog *ctypes.StorageLog) (*transfer, error) {
+	ongLog := types2.Log{
+		Address: ethLog.Address,
+		Topics:  ethLog.Topics,
+		Data:    ethLog.Data,
+	}
+	if ongLog.Address.Hex() == ONG_ADDR {
+		parsed, _ := abi.JSON(strings.NewReader(ERC20ABI))
+		nbc := bind.NewBoundContract(ethcom.Address{}, parsed, nil, nil, nil)
+		tf := new(ERC20Transfer)
+		err := nbc.UnpackLog(tf, "Transfer", ongLog)
+		if err != nil {
+			return nil, err
+		}
+		from, err := common.AddressFromHexString(tf.From.Hex())
+		if err != nil {
+			return nil, err
+		}
+		to, err := common.AddressFromHexString(tf.To.Hex())
+		if err != nil {
+			return nil, err
+		}
+		xfer := &transfer{
+			amount: tf.Value,
+			from:   from,
+			to:     to,
+		}
+		if tf.To.Hex() == GOV_ADDR {
```

Comment:
In `decodeTransfer`, I also check if:

```go
uint64(amount) == info.GasConsumed
```

That is, if the transfer amount matches the `GasConsumed` amount. This is just to catch any edge cases where someone intentionally does a transfer to the Gov address but where it's not actually the transaction fee.

Perhaps do a similar check with `xfer.amount`?

Reply:
The EVM-based ONG transfer is different from the previous one. In the EVM event log, GasConsumed only records the transaction fee for this transfer. If we use EVM to transfer ONG, it will not only cost the handling fee, but also transfer ONG.

Reply:
yes,I had fix